src/NServiceBus.RabbitMQ/Routing/DefaultRoutingKeyConvention.cs

```diff
@@ -23,6 +24,16 @@ static string GetRoutingKey(Type type, string key = "")
 
             var implementedInterface = interfaces.FirstOrDefault();
 
+            if (interfaces.Count > 0)
+            {
+                Logger.Warn("The direct routing topology might not deliver message types with interfaces in their inheritance hierarchy. Message types used for publishing should not implement interfaces.");
+
+                if (interfaces.Count > 1)
+                {
+                    Logger.ErrorFormat("Publishing type has multiple interfaces. Interfaces are {0}, the one that will be used for building routing key is {1}", string.Join(", ", interfaces), implementedInterface);
```

Comment:
Based on my comment above about only having this condition log anything, I think the message should be expanded to specifically point out that because we're only using a single interface, the other interfaces will not participate in the publishing process at all. How about:

>The direct routing topology cannot properly publish a message type that implements more than one relevant interface. The type '{0}' implements the following interfaces: {1}. The interface that will be used is '{2}'. The others will be ignored, and any endpoints that subscribe to those interfaces will not receive a copy of the message.